src/Core/src/Credentials/WebIdentityProvider.php

```diff
@@ -128,14 +128,18 @@ private function getCredentialsFromRole(string $roleArn, string $tokenFile, ?str
      */
     private function getTokenFileContent(string $tokenFile): string
     {
-        if (!file_exists($tokenFile)) {
-            $tokenDir = \dirname($tokenFile);
-            $tokenLink = readlink($tokenFile);
-            clearstatcache(true, $tokenDir . \DIRECTORY_SEPARATOR . $tokenLink);
-            clearstatcache(true, $tokenDir . \DIRECTORY_SEPARATOR . \dirname($tokenLink));
-            clearstatcache(true, $tokenFile);
+        $token = @file_get_contents($tokenFile);
```

Comment:
`@` is to suppress warning in the exact case of hitting the realpath cache with the wrong valued.